packages/openapi-typescript/src/lib/ts.ts

```diff
@@ -308,7 +312,7 @@ export function tsEnumMember(value: string | number, metadata: { name?: string;
       if (invalidCharMatch[0] === name) {
         name = `"${name}"`;
       } else {
-        name = name.replace(JS_PROPERTY_INDEX_INVALID_CHARS_RE, "_");
+        name = name.replace(JS_PROPERTY_INDEX_INVALID_CHARS_RE, (substring) => SPECIAL_CHARACTER_MAP[substring] || "_");
```

Comment:
minor change:

```suggestion
        name = name.replace(JS_PROPERTY_INDEX_INVALID_CHARS_RE, (substring) => (substring in SPECIAL_CHARACTER_MAP) ? SPECIAL_CHARACTER_MAP[substring]: "_");
```

Wouldn’t change functionally how you wrote it, but would allow for having `""` in the `SPECIAL_CHARACTER_MAP` if-needed